dicomexporter/exporter.py

```diff
@@ -0,0 +1,225 @@
+import gzip
+import os
+import shutil
+
+import itk
+import numpy
+from numpy.core.numeric import full
+import vtk
+
+from .dicom import createITKImageReader
+from .itk_utils import convertITKTypeToVTKType, getMetadata, getMetadataList
+# To uncomment when running script manually
+# from helpers.DICOM import createITKImageReader
+# from helpers.itk import convertITKTypeToVTKType, getMetadata, getMetadataList
+# from helpers.volume import VolumeData
+
+
+def convertDICOMVolumeToVTKFile(
+        dicom_directory, 
+        output_file_path,
+        overwrite=False,
+        compress_gzip=True,
+        compress_12_bits=False,
+        blockSize=10 * 1024 * 1024
+    ):
+    """
+    Converts DICOM files in a directory into a VTK file (.vti or .vtkjs)
+    """
+    # Test output_file_path #
+    if not overwrite and os.path.exists(output_file_path):
+        print('Output file already exist', output_file_path, 
+            '\nIf you want to overwrite the file add the \'--overwrite\' flag')
+        return False, None
+
+    itkReader = createITKImageReader(dicom_directory)
+    volume = itkReader.GetOutput() if itkReader is not None else None
+
+    if volume is None:
+        print('Failed to read DICOM volume', dicom_directory, output_file_path)
+        return False, None
+
+    # Test volume data type #
+    ITKType = itk.template(volume)[1][0]
+    volumeDataType = convertITKTypeToVTKType(ITKType)
+
+    if not volumeDataType:
+        print('Data type not handled', ITKType)
+        return False
+
+    # Extract volume parameters #
+    volumeSize = list(volume.GetLargestPossibleRegion().GetSize())
+    volumeOrigin = list(volume.GetOrigin())
+    volumeSpacing = list(volume.GetSpacing())
+    volumeComponents = volume.GetNumberOfComponentsPerPixel()
+    volumeExtent = [
+        0, volumeSize[0] - 1,
+        0, volumeSize[1] - 1,
+        0, volumeSize[2] - 1,
+    ]
+
+    # Extract DICOM fields #
+    if compress_12_bits:
+        bits_stored = getMetadata(itkReader, '0028|0101', int)
+        if bits_stored != 12:
+            print('Data is not 12 bits but', bits_stored)
+            return False

```

Comment:
Should not be treated as an error, it is a valid case (developers will turn on compress_12_bits for ALL their DICOM files, without knowing if those files are 12 or 16 bits).